application/src/main/kotlin/net/corda/v5/application/uniqueness/model/UniquenessCheckError.kt

```diff
@@ -0,0 +1,57 @@
+package net.corda.v5.application.uniqueness.model
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import net.corda.v5.base.annotations.CordaSerializable
+import java.time.Instant
+
+/**
+ * Representation of errors that can be raised by the uniqueness checker. These errors are returned
+ * through the uniqueness client service as well. This representation is agnostic to both the message
+ * bus API and any DB schema that may be used to persist data by the backing store.
+ *
+ * Unfortunately since this is a sealed class with multiple implementations we need to tell Jackson
+ * what types to look for. Hence the annotations. This will be removed when we start using standard
+ * Corda serialization.
+ */
+@CordaSerializable
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
```

Comment:
None of this should be here, this should be invisible to the api user and on the actual implementations in corda-runtime-os. Only interfaces should be exposed here unless the objects are truly trivial POJOs.